broni/shapes/primitives.py

```diff
@@ -0,0 +1,86 @@
+from . import Shape
+from .. import Trajectory
+
+import numpy as np
+
+
+class Sphere(Shape):
+    def __init__(self, x: float, y: float, z: float, r: float):
+        self.center = np.array((x, y, z))
+        self.radius = r
+
+    def intersect(self, trajectory: Trajectory):
+        dist = np.linalg.norm(self.center - trajectory.data, axis=1)
+        return dist <= self.radius
```

Comment:
return np.abs(dist-self.radius) < tolerance

Reply:
not sure this makes sense here

Reply:
Yeah I think I was tired... never mind